js/ui/marker.js

```diff
@@ -85,6 +89,98 @@ Marker.prototype = {
         return this._el;
     },
 
+    /**
+     * Binds a Popup to the Marker
+     * @param {HTMLElement|String|Popup} content the DOM content to appear in the popup, or 
+     *  an instance of the Popup class
+     * @param {Object} [options] options for the Popup class
+     * @param {boolean} [options.closeButton=true] If `true`, a close button will appear in the
+     *   top right corner of the popup.
+     * @param {boolean} [options.closeOnClick=true] If `true`, the popup will closed when the
+     *   map is clicked.
+     * @param {string} options.anchor - A string indicating the popup's location relative to
+     *   the coordinate set via [Popup#setLngLat](#Popup#setLngLat).
+     *   Options are `'top'`, `'bottom'`, `'left'`, `'right'`, `'top-left'`,
+     * `'top-right'`, `'bottom-left'`, and `'bottom-right'`.
+     * @returns {Marker} `this`
+     */
+
+    bindPopup: function(content, options){
```

Comment:
Let's make `Popup` the only type accepted by `bindPopup`, and remove `options`. This separates concerns:

* `Popup` is responsible for construction and configuration of the popup.
* `bindPopup` is responsible for linking it with a `Marker`. It doesn't need to pass through all functionality provided by `Popup`.